cmd/gitserver/server/vcs_syncer_perforce.go

```diff
@@ -70,19 +70,19 @@ func (s *PerforceDepotSyncer) Type() string {
 	return "perforce"
 }
 
-func (s *PerforceDepotSyncer) CanConnect(ctx context.Context, host, username, password string) error {
-	return p4testWithTrust(ctx, host, username, password)
-}
-
 // IsCloneable checks to see if the Perforce remote URL is cloneable.
 func (s *PerforceDepotSyncer) IsCloneable(ctx context.Context, _ api.RepoName, remoteURL *vcs.URL) error {
 	username, password, host, path, err := decomposePerforceRemoteURL(remoteURL)
 	if err != nil {
 		return errors.Wrap(err, "decompose")
 	}
 
+	return isDepotPathCloneable(ctx, host, username, password, path)
+}
+
+func isDepotPathCloneable(ctx context.Context, p4port, p4username, p4passwd, depotPath string) error {
 	// start with a test and set up trust if necessary
-	if err := p4testWithTrust(ctx, host, username, password); err != nil {
+	if err := p4testWithTrust(ctx, p4port, p4username, p4passwd); err != nil {
```

Comment:
Warning: Nit inbound. 😄 
I appreciate making the variable names match the Perforce conventions; let's go all the way and use `p4user`.

```suggestion
	if err := p4testWithTrust(ctx, p4port, p4user, p4passwd); err != nil {
```